<launch>
  <include file="$(find omni_common)/launch/omni_state.launch"/>
  <include file="$(find setting_tf2)/launch/tf2_setting.launch" />


<!--For DoB&Current based position control-->
  <arg name="robot_name"                 default="dasom"/>

  <arg name="position_p_gain"            default="700"/>
  <arg name="position_i_gain"            default="0"/>
  <arg name="position_d_gain"            default="400"/>

  <arg name="dob_p_gain"              default="0"/>
  <arg name="dob_i_gain"              default="0"/>
  <arg name="dob_d_gain"              default="0"/>
  <arg name="dob_polar_moment"        default="0.0027"/>

  <arg name="virtual_mass_x"             default="200.0"/>
  <arg name="virtual_damper_x"           default="10.0"/>
  <arg name="virtual_spring_x"           default="100.0"/>

  <arg name="virtual_mass_y"             default="200.0"/>
  <arg name="virtual_damper_y"           default="10.0"/>
  <arg name="virtual_spring_y"           default="300.0"/>
  
  <arg name="virtual_mass_z"             default="200.0"/>
  <arg name="virtual_damper_z"           default="10.0"/>
  <arg name="virtual_spring_z"           default="300.0"/>

<!--end-->



  <param name="robot_name"               value="$(arg robot_name)"/>

  <param name="position_p_gain"          value="$(arg position_p_gain)"/>
  <param name="position_i_gain"          value="$(arg position_i_gain)"/>
  <param name="position_d_gain"          value="$(arg position_d_gain)"/>

  <param name="virtual_mass_x"           value="$(arg virtual_mass_x)"/>
  <param name="virtual_damper_x"         value="$(arg virtual_damper_x)"/>
  <param name="virtual_spring_x"         value="$(arg virtual_spring_x)"/>

  <param name="virtual_mass_y"           value="$(arg virtual_mass_y)"/>
  <param name="virtual_damper_y"         value="$(arg virtual_damper_y)"/>
  <param name="virtual_spring_y"         value="$(arg virtual_spring_y)"/>

  <param name="virtual_mass_z"           value="$(arg virtual_mass_z)"/>
  <param name="virtual_damper_z"         value="$(arg virtual_damper_z)"/>
  <param name="virtual_spring_z"         value="$(arg virtual_spring_z)"/>

  <param name="cut_off_freq_current"     value="$(arg cut_off_freq_current)"/>


  <node name="torque_jacobian" pkg="dasom_controllers" type="torque_jacobian" required="true" output="screen"> 
    <param name="dob_p_gain"              value="$(arg dob_p_gain)"/>
    <param name="dob_i_gain"              value="$(arg dob_i_gain)"/>
    <param name="dob_d_gain"              value="$(arg dob_d_gain)"/>
    <param name="dob_polar_moment"        value="$(arg dob_polar_moment)"/>
  </node>

</launch>
